glommio/src/io/buffered_file_stream.rs

```diff
@@ -409,7 +409,7 @@ impl StreamWriter {
                         .upgrade()
                         .unwrap()
                         .fdatasync(self.file.as_ref().unwrap().as_raw_fd());
-                    source.add_waiter_single(cx.waker().clone());
+                    source.add_waiter_single(cx.waker());
```

Comment:
@HippoBaro please review this, since we had issues before with `add_waiter_single` vs multiple.